Makefile

```diff
@@ -113,7 +112,7 @@ build: generate fmt vet assets ## Build manager binary.
 run: manifests generate fmt vet assets ## Run a controller from your host.
 	go run ./main.go
 
-docker-build: test ## Build docker image with the manager.
```

Comment:
Why `test` target have been removed from `docker-build`? this for example guarantees on the release pipeline that test pass on the pipeline just before generating all involved images....

Reply:
Because is awfully slow while developing locally. We call the test target explicitly in the `test` workflow in each PR against master so we are not really missing anything in the CI (see https://github.com/3scale-ops/saas-operator/blob/main/.github/workflows/test.yaml).

Reply:
Yes I know, but since the tests runs on the PR test workflow until the release workflow is executed on a github action, it can pass a lot of time (minutes, hours, days).

I think it is also OK to run the test just before generating the final artifact on the same ci-node, ensuring that code is not broken by maybe a lib with floating tag that a developer maintain wrongly (It would be the first time we see a developer not managing correctly their tags...)

Maybe we should just add the test on the release workflow, instead of having it harcoded on the `docker-build`  like we had before? (out of the PR scope btw, we can do it on another PR)

Reply:
No problem with adding the test step to the release workflow, that makes sense.

Reply:
2531271

Reply:
thanks!